django/forms/widgets.py

```diff
@@ -428,6 +431,9 @@ def value_from_datadict(self, data, files, name):
             return False
         return upload
 
+    def use_required_attribute(self, initial):
```

Comment:
I see your point.

If the `BoundField` was passed, then I suppose the entire test could exist in base implementation:

```
    def use_required_attribute(self, bound_field):
        return not self.is_hidden and bound_field.field.required and bound_field.form.use_required_attribute
```

Let me know what you think. I'm happy to move in that direction if you think it is a better design.